### PR TITLE
finished react_todo_app

### PR DESCRIPTION
**Pull request by pushkarskiyrodion**
[DEMO LINK](https://pushkarskiyrodion.github.io/react_todo-app/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,93 +1,200 @@
...
+import { Error } from './components/Error';
+
+import { Todo } from './types/Todo';
+import { PatchTodo } from './types/PatchTodo';
+import { ErrorMessage } from './enums/ErrorMessages';
+import { Filter } from './enums/Filter';
+
+import {
+  USER_ID, deleteTodo, getTodos, patchTodos, postTodos,
+} from './api/todos';
```
Comment (loralevitska) [inline src/App.tsx:16]:
```suggestion
import {
  USER_ID, 
  deleteTodo, 
  getTodos, 
  patchTodos, 
  postTodos,
} from './api/todos';
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,93 +1,200 @@
...
+
+    const fetchData = async () => {
+      try {
+        const todos = await getTodos(USER_ID);
+
+        setTodoList(todos);
+      } catch (err) {
+        setError(ErrorMessage.LOAD);
+      }
+    };
```
Comment (loralevitska) [inline src/App.tsx:42]: Create function out of `useEffect`

File: src/App.tsx [hunk 3]
```diff
@@ -1,93 +1,200 @@
...
+
+  const filteredTodos = useMemo(() => {
+    return todoList.filter(todo => {
+      switch (filterValue) {
+        case Filter.Active:
+          return !todo.completed;
+        case Filter.Completed:
+          return todo.completed;
+        default:
+          return todo;
```
Comment (loralevitska) [inline src/App.tsx:153]:
```suggestion
      switch (filterValue) {
        case Filter.Active:
          return !todo.completed;
          
        case Filter.Completed:
          return todo.completed;
          
        default:
          return todo;
```

File: src/components/NewTodo/NewTodo.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    if (e.code === 'Escape') {
+      setNewTitle(title);
+      onNeedChange(false);
+    }
+  };
+
+  const handleSubmit = async (e: React.BaseSyntheticEvent) => {
+    e.preventDefault();
+
+    if (newTitle === '') {
```
Comment (loralevitska) [inline src/components/NewTodo/NewTodo.tsx:44]:
```suggestion
    if (!newTitle) {
```

**Review by loralevitska — CHANGES_REQUESTED** [review]
Keep up the good work! Some comments to fix:

- don't change footer due to filters. Check yourself according [example](https://todomvc.com/examples/vanillajs/#/completed)
image
- add right condition for 1 item left 
image

**Review by sTorba24 — APPROVED** [review]
Approved!

**Pull request closed without merging**